packages/amplify-category-geo/src/provider-controllers/map.ts

```diff
@@ -120,3 +104,57 @@ export const updateMapResourceWithParams = async (
   printNextStepsSuccessMessage(context);
   return completeParameters.name;
 }
+
+export const removeMapResourceWithParams = async (
+  context: $TSContext,
+  mapToRemove: string,
+  isHeadlessCommand: boolean = false,
+  newDefaultMap?: string
+): Promise<string> => {
+  const { amplify } = context;
+  const mapParams = await getCurrentMapParameters(mapToRemove);
+  try {
+    await amplify.removeResource(context, category, mapToRemove, { headless: isHeadlessCommand })
+    .then(async (resource: { service: string; resourceName: string } | undefined) => {
+      // choose another default if removing a default map
+      if (resource?.service === ServiceName.Map && mapParams.isDefault) {
+        // select new default map in headless mode
+        if (isHeadlessCommand) {
+          const remainingMaps = ((await context.amplify.getResourceStatus()).allResources as any[])
```

Comment:
Don't we need to filter the current map being removed also to get the remaining maps or at this point the current map info is already deleted from `amplify-meta` ?

Reply:
The current map is already deleted from meta